File: CodingSamples/Java/Language/EnumTypeTest/Investment.java
enum RiskLevel {
	NONE, LOW, HIGH;
}

class Investment {

	private double installment;
	private int years;
	private RiskLevel risk;

	public Investment(double amount, int period) {
		installment = amount;
		years = period;
		risk = RiskLevel.NONE;
	}

	public void allowRisk(boolean yes) {
		risk = yes ? RiskLevel.LOW : RiskLevel.NONE;
	}

	//method overloading - defining multiple methods in a given class
	//with same name but different list of parameter types
	public void allowRisk(RiskLevel level) {
		risk = level;
	}

	public double totalAmount() {
		return installment * years;
	}

	public double futureValue() {
		float i;
		switch(risk){
			case LOW:
				i = 0.11f;
				break;
			case HIGH:
				i = 0.15f;
				break;
			default:
				i = 0.06f;
		}
		return (installment / i) * (Math.pow(1 + i, years) - 1);
	}

}

File: CodingSamples/DotNet/Language/EnumTypeTest/DemoApp/Investment.cs
enum RiskLevel {None, Low, High}

struct Investment
{
    private double installment;
    private int period;
    private RiskLevel risk;

    public Investment(double amount, int years)
    {
        installment = amount;
        period = years;
        risk = RiskLevel.None;
    }

    public void AllowRisk(bool yes)
    {
        risk = yes ? RiskLevel.Low : RiskLevel.None;
    }

    //method overloading - defining multiple methods in
    //a type with same name but different list of 
    //parameter types
    public void AllowRisk(RiskLevel level)
    {
        risk = level;
    }

    public double TotalAmount()
    {
        return installment * period;
    }

    public double FutureValue()
    {
        float i;
        switch(risk)
        {
            case RiskLevel.Low:
                i = 0.11f;
                break;
            case RiskLevel.High:
                i = 0.15f;
                break;
            default:
                i = 0.06f;
                break;
        }
        return (installment / i) * (Math.Pow(1 + i, period) - 1);
    }
}
